component/security.php

```diff
@@ -0,0 +1,170 @@
+        $numberstreet = htmlspecialchars($numberstreet);
+        if (!preg_match('~[0-9]{5}~', $postalCode)) //strlen($postalCode) !== 5 || !is_int($postalCode)
+        {
+            return false;
+        }
+        else if (
+            !is_string($street) || 
+            strlen($street) < 2 || 
+            strlen($street) > 50) 
+        {
+            return false;
+        }
+        else if (
+            !is_string($town) || 
+            strlen($town) < 2 || 
+            strlen($town) > 50) 
+        {
+            return false;
+        }
+        else {         
+            return $street . $numberstreet;
+        }
+    }
+
+    public static function testCheckObligate($value){
+        if (htmlspecialchars($value) === "on"){
+            return true;
+        }
+        else return false;
+    }
+
+    public static function testBirthday($birthDate){
+        $birthDate = htmlspecialchars($birthDate); // OKAY
+        $array = explode("-",$birthDate);
+        if (
+            is_string($birthDate) && 
+            strlen($array[0]) === 4 && 
+            strlen($array[1]) === 2 && 
+            strlen($array[2]) === 2
+        )
+        {
+            // Test date
+            $format = "Y-m-d";
+            $date_naissance = DateTime::createFromFormat($format, $birthDate); // Créer une date avec la date de naissance
+            $date = new DateTime(); // Créer une date avec la date d'aujourd'hui
+            $date = $date->sub(new DateInterval('P18Y')); // On remplace la date en retirant 18 ans
+            if($date_naissance >= $date) // On compare les deux dates pour verifier la majorité de 18 ans
+            {
+                return false;
+            }
+            else return $birthDate;
+        }
+        else return false;
+    }
+    public static function testAmount($amount){
+        $amount = htmlspecialchars($amount);
+       if(!is_numeric($amount))
+       {
+           return false;
+       }
+       else return $amount;
+    }
+
+    public static function testGender($value){
+        if($value === 'man' || $value === 'woman'){
+            return htmlspecialchars($value);
+        } else {
+            return false;
+        }
+    }
+
+    public static function testUploadedFile($file){
+        if
+        (
+            ($file['type'] === 'image/png' || 'image/jpeg' || 'application/pdf') && 
+            ($file['size'] <= 1000000) && 
+            ($file['error'] === UPLOAD_ERR_OK)
+        )
+        {
+            return $file;
+        } 
+        else 
+        {
+            return false;
+        }
+    }
+
+    public static function testID($value){
+        if(preg_match_all('/[0-9]/', $value) !== strlen($value)){
+            return false;
+        }
+        else return htmlspecialchars($value);
+    }
+
+    public static function generateToken($userid, $type){
+        $privatekey = "hjfeo78scds";
+        $privatekeye = hash('sha256', $privatekey);
```

Comment:
The private key being so small and the hash method being sha256 make this token not very secure